### ISSUE TITLE
Download tab not working for when selecting 10 or more tcga pancan studies

### ISSUE THREAD

**Issue by Sjoerd-van-Hagen** [cBioPortal/cbioportal#7801]
User reported error in the Google group: https://groups.google.com/u/1/g/cbioportal/c/uypMeQKYJuw/m/WDxUBwVEBQAJ

I tested with MSK impact to see if it was the number of samples, but that worked. It also worked with the first four pancan atlas studies. When selecting the first 10 it stopped working. See below of the URL.

https://www.cbioportal.org/results/download?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_list=acc_tcga_pan_can_atlas_2018%2Cchol_tcga_pan_can_atlas_2018%2Cblca_tcga_pan_can_atlas_2018%2Ccoadread_tcga_pan_can_atlas_2018%2Cbrca_tcga_pan_can_atlas_2018%2Clgg_tcga_pan_can_atlas_2018%2Cgbm_tcga_pan_can_atlas_2018%2Ccesc_tcga_pan_can_atlas_2018%2Cesca_tcga_pan_can_atlas_2018%2Cstad_tcga_pan_can_atlas_2018&case_set_id=all&data_priority=0&gene_list=TP53&geneset_list=%20&profileFilter=0&tab_index=tab_visualize

Error message in console (Firefox): Uncaught (in promise) Error: Request has been terminated
Possible causes: the network is offline, Origin is not allowed by Access-Control-Allow-Origin, the page is being unloaded, etc.

Request timeout on the server perhaps, because it is taking a lot of time.

After fixing, please respond to the Google group issue as well.

**Comment by dippindots** [comment]
@inodb I checked the backend implementation of the `geneirc-assay-data` endpoint, and I haven't seen an obvious code that leads to the slowness. 
Then I realized that for other `genetic data`, we are allow user to download `queried genes * number of samples` data at a time. For example, if user queried 100 genes (reached queried genes limitation), then for 10000 samples set, the download will need to fetch `100 x 10000 data` from the backend.
But for `microbiome data` in `tcga`, we have around 1400 entities, and 10000 samples, then we need to fetch `1400 * 10000 data` from the backend at a time, I think this is why it's causing the crush.

**Comment by dippindots** [comment]
We proposed to solve this problem by only download selected entities in `oncoprint heatmap`. That will reduce lots of unnecessary data to be fetched from the backend.

But that would require multiple studies support on `OncoPrint heatmap`, we need to finish this ticket first: https://github.com/cBioPortal/cbioportal/issues/7850.

So, currently, we will disable the generic assay download for now. And enable download again after `OncoPrint heatmap` has been implemented.
